apps/comps/components/featured-competition/top-leaders-list.tsx

```diff
@@ -0,0 +1,84 @@
+import Link from "next/link";
+import React from "react";
+
+import { Skeleton } from "@recallnet/ui2/components/skeleton";
+
+import { AgentCompetition } from "@/types";
+
+import { AgentAvatar } from "../agent-avatar";
+import { RankBadge } from "../agents-table/rank-badge";
+
+interface TopLeadersListProps {
+  agents: AgentCompetition[];
+  isLoading: boolean;
+}
+
+export const TopLeadersList: React.FC<TopLeadersListProps> = ({
+  agents,
+  isLoading,
+}) => {
+  if (isLoading) {
+    return (
+      <div className="grid grid-cols-4 gap-4 rounded-lg border-y bg-[#050507] p-3">
+        {Array.from({ length: 3 }).map((_, i) => (
+          <div key={i} className="col-span-4 flex items-center gap-4">
+            <Skeleton className="h-8 w-[62px] rounded-md" />
+            <div className="flex items-center gap-3">
+              <Skeleton className="h-10 w-10 rounded-full" />
+              <Skeleton className="h-5 w-24" />
+            </div>
+            <Skeleton className="h-5 w-28" />
+            <Skeleton className="h-5 w-24" />
+          </div>
+        ))}
+      </div>
+    );
+  }
+
+  if (agents.length === 0) {
+    return (
+      <div className="flex h-[64px] items-center justify-center rounded-lg bg-gray-900/50 p-3">
+        <p className="text-gray-400">No participants yet.</p>
+      </div>
+    );
+  }
+
+  return (
+    <div className="flex flex-col gap-4 rounded-lg border-y bg-[#050507] p-3">
+      {agents.map((agent) => (
+        <div
+          key={agent.id}
+          className="hover:bg-card col-span-4 grid grid-cols-[70px_1fr_auto_auto] gap-4"
+        >
+          <RankBadge position={agent.position} />
+          <Link
+            href={`/agents/${agent.id}`}
+            className="flex items-center gap-3"
+          >
+            <AgentAvatar agent={agent} size={32} />
```

Comment:
this is causing a hydration error: `In HTML, <a> cannot be a descendant of <a>.
This will cause a hydration error.`
<img width="960" alt="Screenshot 2025-06-12 at 3 07 29 PM" src="https://github.com/user-attachments/assets/9ed7802b-2e1b-43f3-8299-cbf519ada09c" />

Reply:
fixed in [9ed2821](https://github.com/recallnet/js-recall/pull/570/commits/9ed28212509c2fcb81ea4d84635a74b302c8a86b)